src/sagas/repository.ts

```diff
@@ -1023,6 +1028,40 @@ export function* rejectSnapshotAccessRequest({ payload }: any): any {
   }
 }
 
+export function* approveSnapshotAccessRequest({ payload }: any): any {
+  const {
+    snapshotAccessRequestResponse: { id, snapshotName },
+  } = payload;
+  const snapshotRequest: SnapshotRequestModel = {
+    name: generateSnapshotNameFromAccessRequestInformation(id, snapshotName),
+    description: `Snapshot created from ${snapshotName}`,
+    contents: [
+      {
+        datasetName: 'unused',
+        mode: SnapshotRequestContentsModelModeEnum.ByRequestId,
+        requestIdSpec: {
+          snapshotRequestId: id,
+        },
+      },
+    ],
+  };
+  try {
+    yield call(authPut, `/api/repository/v1/snapshotAccessRequests/${id}/approve`);
+    yield call(authPost, '/api/repository/v1/snapshots', snapshotRequest);
+    yield put({
+      type: ActionTypes.APPROVE_SNAPSHOT_ACCESS_REQUEST_SUCCESS,
+    });
+    yield put({
+      type: ActionTypes.GET_SNAPSHOT_ACCESS_REQUESTS,
```

Comment:
why get snapshot access requests here?

Reply:
We want to refresh the page so it gets the new status

Reply:
Hmmm... Yeah, I think that makes sense. I'll add it.